### PR TITLE
fix(macos): minor tweaks to for cargo build to run successfully

### PR DESCRIPTION
**Pull request by pims**
I understand MacOS isn't a supported target for any kind of production workload.
But being able to run locally without needing to build a new Docker image has its benefits.

I'm extremely new at Rust, so let me know if what I did doesn't make sense.
No worries if you decide that this is not something you'd like to merge, but it's been very useful for me to iterate quickly while exploring different configuration settings.

### CODE REVIEW

File: bin/src/main.rs [hunk 1]
```diff
@@ -292,7 +292,7 @@ fn update_process_limits(config: &Config) -> Result<(), anyhow::Error> {
 }
 
 #[cfg(not(target_os = "linux"))]
-fn update_process_limits(_: &Config) -> Result<(), StartupError> {
+fn update_process_limits(_: &Config) -> Result<(), anyhow::Error> {
```
Comment (Keksoj) [inline bin/src/main.rs:295]: It seems you've caught up a piece of code we had forgotten while updating the return types everywhere!

File: bin/src/worker.rs [hunk 2]
```diff
@@ -322,30 +323,11 @@ extern "C" {
...
-        } else {
-            panic!();
-        }
-    } else {
-        panic!("buffer too small");
+pub unsafe fn get_executable_path() -> anyhow::Result<String> {
+    let path = env::current_exe()?;
+    match path.into_os_string().into_string() {
+        Ok(exe_path) => Ok(exe_path),
+        Err(_) => bail!("Failed to convert PathBuf to String"),
```
Comment (Keksoj) [inline bin/src/worker.rs:330]: this is very consistent with the rest of Sōzu's code and looks more straightforward.

**Review by Keksoj — APPROVED** [review]
This looks very good. You have stumbled upon an unsurprising weakness, because we never build Sōzu for MacOS, so we never run this code.

Does it work on your machine now?

I'd still like to have @FlorentinDUBOIS 's opinion on this but I'm down to merge this.

Thank you! :pray:

File: bin/src/worker.rs [hunk 3]
```diff
@@ -322,30 +323,11 @@ extern "C" {
...
-            path.to_str()
-                .expect("failed to convert CString to String")
-                .to_string()
-        } else {
-            panic!();
-        }
-    } else {
-        panic!("buffer too small");
+pub unsafe fn get_executable_path() -> anyhow::Result<String> {
+    let path = env::current_exe()?;
```
Comment (FlorentinDUBOIS) [inline bin/src/worker.rs:327]: could you add `.with_context(|| "failed to retrieve current executable path")?`

File: bin/src/worker.rs [hunk 4]
```diff
@@ -322,30 +323,11 @@ extern "C" {
...
+    match path.into_os_string().into_string() {
```
Comment (FlorentinDUBOIS) [inline bin/src/worker.rs:328]: As we come from a `PathBuf`, we could use the `to_string_lossy` method and remove the `match`

```suggestion
    path.to_string_lossy().to_string()
```

**Review by FlorentinDUBOIS — CHANGES_REQUESTED** [review]

File: bin/src/worker.rs [hunk 5]
```diff
@@ -322,30 +323,11 @@ extern "C" {
...
-            let path = CString::from_raw(ptr2);
-            path.to_str()
-                .expect("failed to convert CString to String")
-                .to_string()
-        } else {
-            panic!();
-        }
-    } else {
-        panic!("buffer too small");
+pub unsafe fn get_executable_path() -> anyhow::Result<String> {
```
Comment (FlorentinDUBOIS) [inline bin/src/worker.rs:326]: we could remove the `unsafe` keyword here

Comment (pims) [reply bin/src/worker.rs:326]: I've kept `unsafe` to keep it consistent with other platforms and not change the call site. Do you suggest I remove it just for `macOS` ?

**Comment by pims** [comment]
> This looks very good. You have stumbled upon an unsurprising weakness, because we never build Sōzu for MacOS, so we never run this code.
> 
> Does it work on your machine now?
> 
> I'd still like to have @FlorentinDUBOIS 's opinion on this but I'm down to merge this.
> 
> Thank you! 🙏

@Keksoj it does :)
Here's the build output:
```
tim@laptop sozu % cargo build
warning: unused imports: `ErrorKind`, `Error`, `os::unix::io::AsRawFd`
 --> lib/src/metrics/writer.rs:4:16
  |
4 |     io::{self, Error, ErrorKind, Write},
  |                ^^^^^  ^^^^^^^^^
5 |     net::SocketAddr,
6 |     os::unix::io::AsRawFd,
  |     ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused imports: `c_uint`, `c_void`, `iovec`, `msghdr`
 --> lib/src/metrics/writer.rs:9:12
  |
9 | use libc::{c_uint, c_void, iovec, msghdr};
  |            ^^^^^^  ^^^^^^  ^^^^^  ^^^^^^

warning: `sozu-lib` (lib) generated 2 warnings
   Compiling sozu v0.14.2 (/Users/tim/code/rust/sozu/bin)
warning: unused import: `std::ffi::CString`
 --> bin/src/worker.rs:4:5
  |
4 | use std::ffi::CString;
  |     ^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `std::iter::repeat`
 --> bin/src/worker.rs:6:5
  |
6 | use std::iter::repeat;
  |     ^^^^^^^^^^^^^^^^^

warning: unused import: `std::ptr::null_mut`
  --> bin/src/worker.rs:12:5
   |
12 | use std::ptr::null_mut;
   |     ^^^^^^^^^^^^^^^^^^

warning: unused import: `bail`
  --> bin/src/worker.rs:22:14
   |
22 | use anyhow::{bail, Context};
   |              ^^^^

warning: unused import: `PATH_MAX`
  --> bin/src/worker.rs:25:20
   |
25 | use libc::{c_char, PATH_MAX};
   |                    ^^^^^^^^

warning: unused import: `bail`
  --> bin/src/main.rs:41:14
   |
41 | use anyhow::{bail, Context};
   |              ^^^^

warning: `sozu` (bin "sozu") generated 6 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 2.55s
```

File: bin/src/worker.rs (hunk 5 above)
Comment (Keksoj) [reply bin/src/worker.rs:326]: yes ! remove if possible!

**Comment by Keksoj** [comment]
> > Does it work on your machine now?

> @Keksoj it does :) Here's the build output:

@pims great! Don't remove the unused imports, as they are used in other architectures ;-)

File: bin/src/worker.rs (hunk 5 above)
Comment (pims) [reply bin/src/worker.rs]: Done.

**Review by FlorentinDUBOIS — APPROVED** [review]

**Comment by FlorentinDUBOIS** [comment]
Thank you a lot for the pull request <3

**Pull request merged**